src/rgw/store/simplefile/writer.cc

```diff
@@ -0,0 +1,161 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t
+// vim: ts=8 sw=2 smarttab ft=cpp
+/*
+ * Ceph - scalable distributed file system
+ * Simple filesystem SAL implementation
+ *
+ * Copyright (C) 2022 SUSE LLC
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ */
+#include <memory>
+#include "rgw_sal.h"
+#include "rgw_sal_simplefile.h"
+#include "store/simplefile/bucket.h"
+#include "store/simplefile/writer.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+using namespace std;
+
+namespace rgw::sal {
+
+SimpleFileAtomicWriter::SimpleFileAtomicWriter(
+  const DoutPrefixProvider *_dpp,
+  optional_yield _y,
+  std::unique_ptr<rgw::sal::Object> _head_obj,
+  SimpleFileStore *_store,
+  const rgw_user& _owner,
+  const rgw_placement_rule *_ptail_placement_rule,
+  uint64_t _olh_epoch,
+  const std::string &_unique_tag
+) : Writer(_dpp, _y), store(_store),
+    obj(_store, _head_obj->get_key(), _head_obj->get_bucket()),
+    owner(_owner),
+    placement_rule(_ptail_placement_rule), olh_epoch(_olh_epoch),
+    unique_tag(_unique_tag), bytes_written(0) {
+
+  lsfs_dout(dpp, 10) << "head_obj: " << _head_obj->get_key()
+                     << ", bucket: " << _head_obj->get_bucket()->get_name()
+                     << dendl;
+}
+
+int SimpleFileAtomicWriter::prepare(optional_yield y) {
+  lsfs_dout(dpp, 10) << ": unimplemented, return success." << dendl;
+  // TODO: create meta file for this new object
+  return 0;
+}
+
+int SimpleFileAtomicWriter::process(bufferlist &&data, uint64_t offset) {
+  lsfs_dout(dpp, 10) << "data len: " << data.length()
+                     << ", offset: " << offset << dendl;
+
```

Comment:
Thinking about performance and given this is called from a loop and the writer instance is the same....
Maybe we could reuse the ofstream instance and even avoid the seekp call, flush and close.
Could we just call flush and close in the complete method?

Also avoid recalculating the object_path, as it's going to be the same for every call in the loop.